lib/ecto/query/builder/filter.ex

```diff
@@ -65,6 +65,7 @@ defmodule Ecto.Query.Builder.Filter do
   end
 
   def escape(kind, expr, vars, env) when is_list(expr) do
+    assert_no_nils!(expr, Ecto.Query.CompileError)
```

Comment:
Ha, ok I had that...should have commited it. Then I thought I should just reuse.